src/service/resources-impl.js

```diff
@@ -239,43 +279,157 @@ export class ResourcesImpl {
     }
   }
 
+  /** @override */
+  isExperimentOn(name) {
+    if (name == 'intersect-resources') {
+      return !!this.intersectionObserver_;
+    }
+    return false;
+  }
+
+  /**
+   * @param {!Array<!IntersectionObserverEntry>} entries
+   * @param {!IntersectionObserver} unusedObserver
+   * @private
+   */
+  intersects_(entries, unusedObserver) {
+    dev().fine(TAG_, 'intersect', entries);
+
+    const toUnload = [];
+
+    const promises = entries.map(entry => {
```

Comment:
Note that there may be later entries with more up-to-date info on the same element. We may need to work from right to left, and skip the leftmost entires that are old.

Reply:
Can this actually happen in practice? I'd be surprised.